workflows/seqinspector.nf

```diff
@@ -50,45 +57,163 @@ workflow SEQINSPECTOR {
         Channel.fromPath(params.multiqc_config, checkIfExists: true) :
-        Channel.empty()
-    ch_multiqc_logo          = params.multiqc_logo ?
+        Channel.fromPath("$projectDir/assets/multiqc_config.yml", checkIfExists: true)
+    ch_multiqc_logo   = params.multiqc_logo ?
         Channel.fromPath(params.multiqc_logo, checkIfExists: true) :
         Channel.empty()
 
-    summary_params      = paramsSummaryMap(
+    summary_params                        = paramsSummaryMap(
         workflow, parameters_schema: "nextflow_schema.json")
-    ch_workflow_summary = Channel.value(paramsSummaryMultiqc(summary_params))
-
+    ch_workflow_summary                   = Channel.value(
+        paramsSummaryMultiqc(summary_params))
     ch_multiqc_custom_methods_description = params.multiqc_methods_description ?
         file(params.multiqc_methods_description, checkIfExists: true) :
         file("$projectDir/assets/methods_description_template.yml", checkIfExists: true)
     ch_methods_description                = Channel.value(
         methodsDescriptionText(ch_multiqc_custom_methods_description))
 
-    ch_multiqc_files = ch_multiqc_files.mix(
+    ch_multiqc_extra_files = ch_multiqc_extra_files.mix(
         ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))
-    ch_multiqc_files = ch_multiqc_files.mix(ch_collated_versions)
-    ch_multiqc_files = ch_multiqc_files.mix(
+    ch_multiqc_extra_files = ch_multiqc_extra_files.mix(ch_collated_versions)
+    ch_multiqc_extra_files = ch_multiqc_extra_files.mix(
         ch_methods_description.collectFile(
             name: 'methods_description_mqc.yaml',
             sort: true
         )
     )
 
     MULTIQC (
-        ch_multiqc_files.collect(),
+        ch_multiqc_files
+            .map { meta, file -> file }
+            .mix(ch_multiqc_extra_files)
+            .collect(),
+        ch_multiqc_config.toList(),
+        Channel.empty().toList(),
+        ch_multiqc_logo.toList()
+    )
+
+    // Generate reports by lane
+    multiqc_extra_files_per_lane = ch_multiqc_files
+        .filter { meta, sample -> meta.lane }
+        .map { meta, sample -> meta.lane }
+        .unique()
+        .map { lane -> [lane:lane] }
+        .combine(ch_multiqc_extra_files)
+
+    lane_mqc_files = ch_multiqc_files
+        .filter { meta, sample -> meta.lane }
+        .mix(multiqc_extra_files_per_lane)
+        .map { meta, sample -> [ "[LANE:${meta.lane}]", meta, sample ] }
+        .groupTuple()
+        .tap { mqc_by_lane }
+        .collectFile{
+            lane, meta, samples -> [
+                "${lane}_multiqc_extra_config.yml",
+                """
+                    |output_fn_name: \"${lane}_multiqc_report.html\"
+                    |data_dir_name:  \"${lane}_multiqc_data\"
+                    |plots_dir_name: \"${lane}_multiqc_plots\"
+                """.stripMargin()
+            ]
+        }
+        .map { file -> [ (file =~ /(\[LANE:.+\])/)[0][1], file ] }
```

Comment:
I'm working on this now but it's taking some time, fighting against nextflow error messages again...

Reply:
I've continued a bit with this but didn't have time to get very far. I still think it would be nice to refactor this to avoid duplicating the logic three times. I think I'll make a new PR for this once it is ready, as @MatthiasZepper suggested.